app/ui-message/client/messageBox/userActionIndicator.js

```diff
@@ -0,0 +1,49 @@
+import { Template } from 'meteor/templating';
+
+import { UserAction, USER_TYPING, USER_RECORDING, USER_UPLOADING } from '../../../ui';
+import { t } from '../../../utils';
+import { getConfig } from '../../../ui-utils/client/config';
+import './userActionIndicator.html';
+
+const maxUsernames = parseInt(getConfig('max-usernames-typing')) || 4;
+
+const showUserActivity = (activity, id) => {
+	const action = activity.split('-')[1];
+	const users = UserAction.get(id, activity);
+
+	if (users.length === 0) {
+		return;
+	}
+	if (users.length === 1) {
+		return {
+			action,
+			multi: false,
+			selfActivity: UserAction.selfActivity[activity],
+			users: users[0],
+		};
+	}
+	let last = users.pop();
+	if (users.length >= maxUsernames) {
+		last = t('others');
+	}
+	let usernames = users.slice(0, maxUsernames - 1).join(', ');
+	usernames = [usernames, last];
+	return {
+		action,
+		multi: true,
+		selfActivity: UserAction.selfActivity[activity],
+		users: usernames.join(` ${ t('and') } `),
+	};
+};
+
+Template.userActionIndicator.helpers({
+	typingUsersData() {
```

Comment:
I believe you can simplify this and return the final strings here instead of parameters to let the template create the string. So the template just mount the `<strong>names</string>activity[, <strong>names</string>activity]`

Example of final string that would be shown: 
**rodrigo, rafael and gabriel** are typing, **marcelo** is recording...

Example of data to be returned here to the template:
```
[{
  users: 'rodrigo, rafael and gabriel',
  activity: 'are typing'
}, {
  users: 'marcelo',
  activity: 'is recording'
}]
```